cmd/gaia/test_cli/cli_test.go

```diff
@@ -0,0 +1,290 @@
+	gaiacliHome := fmt.Sprintf("%s%s%s%s.test_gaiacli", tmpDir, string(os.PathSeparator), name, string(os.PathSeparator))
+	if _, err := os.Stat(gaiacliHome); os.IsNotExist(err) {
+		os.MkdirAll(gaiacliHome, os.ModePerm)
+	}
+	if _, err := os.Stat(gaiadHome); os.IsNotExist(err) {
+		os.MkdirAll(gaiadHome, os.ModePerm)
+	}
+	return gaiadHome, gaiacliHome
+}
+
+func InitializeFixtures(t *testing.T) (f Fixtures) {
+	f = NewFixtures(t)
+
+	// Run gaiad unsafe-reset-all and remove the old config, gentx files
+	f.GDUnsafeResetAll()
+	os.RemoveAll(filepath.Join(f.GaiadHome, "config", "gentx"))
+
+	// Delete old keys if they exist
+	f.GCLIKeysDelete(keyFoo, "--force")
+	f.GCLIKeysDelete(keyBar, "--force")
+
+	// Add the keys back in
+	f.GCLIKeysAdd(keyFoo)
+	f.GCLIKeysAdd(keyBar)
+
+	// Initalize gaiad
+	f.GDInit()
+
+	// Add genesis account
+	f.GDAddGenesisAccount(f.GCLIKeysAddress(keyFoo), fooCoins)
+
+	// Generate bonding transaction
+	f.GDGenTx(keyFoo)
+
+	// Collect gentxs and finalize genesis
+	f.GDCollectGenTxs()
+
+	return
+}
+
+func (f Fixtures) GDStart(flags ...string) {
+	buf := f.executeDaemon(flags, "start", f.gdHome(), "--rpc.laddr", f.RPCAddr, "--p2p.laddr", f.P2PAddr)
+	if buf.Len() > 0 {
+		f.T.Log("START", buf.String())
+	}
+}
+
+func (f Fixtures) GDUnsafeResetAll(flags ...string) {
+	buf := f.executeDaemon(flags, "unsafe-reset-all", f.gdHome())
+	if buf.Len() > 0 {
+		f.T.Log("UNSAFE_RESET_ALL", buf.String())
+	}
+}
+
+func (f Fixtures) GDCollectGenTxs(flags ...string) {
+	buf := f.executeDaemon(flags, "collect-gentxs", f.gdHome())
+	if buf.Len() > 0 {
+		f.T.Log("COLLECT_GENTXS", buf.String())
+	}
+}
+
+func (f Fixtures) GDGenTx(name string, flags ...string) {
+	buf := f.executeDaemon(flags, "gentx", "--name", name, "--home-client", f.GaiaCliHome, f.gdHome())
+	if buf.Len() > 0 {
+		f.T.Log("GENTX", buf.String())
+	}
+	_, err := os.Stdin.Write([]byte(app.DefaultKeyPass))
+	require.NoError(f.T, err)
+}
+
+func (f Fixtures) GDAddGenesisAccount(account sdk.AccAddress, coins sdk.Coins, flags ...string) {
+	buf := f.executeDaemon(flags, "add-genesis-account", string(account), coins.String(), f.gdHome())
+	if buf.Len() > 0 {
+		f.T.Log("ADD_GENESIS_ACCOUNT", buf.String())
+	}
+}
+
+func (f Fixtures) GDInit(flags ...string) {
+	buf := f.executeDaemon(flags, "init", "--moniker", keyFoo, f.gdHome())
+
+	// Parse response and set chainID
+	var chainID string
+	var initRes map[string]json.RawMessage
+	err := json.Unmarshal(buf.Bytes(), &initRes)
+	require.NoError(f.T, err)
+	err = json.Unmarshal(initRes["chain_id"], &chainID)
+	require.NoError(f.T, err)
+	f.ChainID = chainID
+}
+
+func (f Fixtures) GCLIKeysDelete(name string, flags ...string) {
+	_, err := f.executeCLIWError(flags, "keys", "delete", name, f.gcliHome())
+	if err.Error() != fmt.Sprintf("Key %s not found", name) {
+		require.NoError(f.T, err)
+	}
+}
+
+func (f Fixtures) GCLIKeysAdd(name string, flags ...string) {
```

Comment:
This is the function I'm trying to get to work. If you run `go test -v cmd/gaia/test_cli/cli_test.go` you should see the errors here.